modules/deployments-comparing-deploymentconfigs.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * applications_and_projects/deployments/what-deployments-are.adoc
+
+[id='deployments-comparing-deploymentconfigs-{context}']
+= Comparing Deployments and DeploymentConfigs
```

Comment:
Also an important difference is the different properties of CAP theorem those 2 designs have chosen. DCs prefer consistency so deployer pod managing a rollout has identity and if a node running this pod goes down or network gets partitioned it won't get replaced but the process will wait till the node comes back up or the node object gets (manually deleted) also deleting the pod. What this means is that you cannot delete the pod to unstuck the rollout, pods are deleted by the kubelet but that on the inaccessible node. On the other hand Deployments are being driven from controller manager which is running in HA mode on masters and uses leader election (takes availability over consistency) your rollouts won't get stuck. AS it can happen with leader election and time skews in case of failures more masters could be acting on the Deployment at the same time but it should get reconciled shortly.

Also the general sentiment is that we would like users to choose Deployments by default and go for DCs only if the need a specific feature - so lifecycle hooks, automatic rollback or custom deployments.

Reply:
@tnozicka What if I replaced the whole first few paragraphs with something like the following?

"Both Kubernetes Deployments and OpenShift Container Platform-provided DeploymentConfigs are supported in OpenShift Container Platform; however, you should use Deployments unless you need a specific feature provided by DeploymentConfigs.

The following sections go into more details on the differences between the two object types to further help you decide which type to use."

Reply:
@tnozicka As for the important difference that you mention above, we could potentially cover that in the middle of the intro paragraph I proposed above, but I'm wondering if we could describe it more generically, without going into the examples. What I have so far is this:

"One important difference between Deployments and DeploymentConfigs is that DeploymentConfigs prefer consistency, whereas Deployments prefer availability." 

To add on to this, is there a less specific way that you could describe how DeploymentConfigs prefer consistency vs. Deployments prefer availability?

Reply:
> "Both Kubernetes Deployments and OpenShift Container Platform-provided DeploymentConfigs are supported in OpenShift Container Platform; however, you should use Deployments unless you need a specific feature provided by DeploymentConfigs.
The following sections go into more details on the differences between the two object types to further help you decide which type to use."

`s/specific feature/specific feature or behaviour/`

`you should use Deployments` sounds a bit strong to me given this is an opinionated suggestion, maybe `we'd advise you to`?

sounds good otherwise

Reply:
> "One important difference between Deployments and DeploymentConfigs is that DeploymentConfigs prefer consistency, whereas Deployments prefer availability."

Unfortunately, this doesn't say the same thing and I don't think people would understand what it means.

> To add on to this, is there a less specific way that you could describe how DeploymentConfigs prefer consistency vs. Deployments prefer availability?

yeah, this is pretty advanced topic, surely it could be phrased better, but I am not sure people will understand the meaning without explaining it in detail.